### PR TITLE
cli: Fix cli using memory sorter when working on 4.0.x cluster

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- Unified Sorter has become the default on 4.0.13 for the 4.0.X branch, but the current cli treats all 4.0.X versions as a whole and disables Unified Sorter by default.

Close https://github.com/pingcap/ticdc/issues/2373

### What is changed and how it works?
- Better version discrimination.

### Check List 

Tests 

 - Unit test

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch

### Release note

```release-note
Fix CLI back-compatibility
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
LGTM

**Comment by overvenus** [comment]
Does it need to be cherry-picked to release-4.0?

File: pkg/version/check.go [hunk 1]
```diff
@@ -165,7 +191,7 @@ func GetTiCDCClusterVersion(captureInfos []*model.CaptureInfo) (TiCDCClusterVers
 		if captureInfo.Version != "" {
 			ver, err = semver.NewVersion(removeVAndHash(captureInfo.Version))
 		} else {
-			ver = semver.New("4.0.1")
+			ver = MinTiCDCVersion
```
Comment (overvenus) [inline pkg/version/check.go:194]: Can it be TiCDCClusterVersionUnknown?

File: cmd/client_changefeed.go [hunk 2]
```diff
@@ -270,14 +270,19 @@ func verifyChangefeedParameters(ctx context.Context, cmd *cobra.Command, isCreat
...
-		}
-		log.Warn("The TiCDC cluster is built from 4.0-release branch, the old-value and unified-sorter are disabled by default.")
+		log.Warn("The TiCDC cluster is built from an older version, disabling old value by default.",
+			zap.String("version", cdcClusterVer.String()))
+	}
+
+	sortEngineFlag := cmd.Flag("sort-engine")
+	if !sortEngineFlag.Changed && !cdcClusterVer.ShouldEnableUnifiedSorterByDefault() {
+		sortEngine = model.SortInMemory
+		log.Warn("The TiCDC cluster is built form an older version, disabling Unified Sorter by default",
```
Comment (amyangfei) [inline cmd/client_changefeed.go:282]:
```suggestion
		log.Warn("The TiCDC cluster is built from an older version, disabling Unified Sorter by default",
```

File: pkg/version/check.go [hunk 3]
```diff
@@ -144,14 +147,37 @@ func CheckStoreVersion(ctx context.Context, client pd.Client, storeID uint64) er
...
+// ShouldEnableUnifiedSorterByDefault returns whether Unified Sorter should be enabled by default
+func (v *TiCDCClusterVersion) ShouldEnableUnifiedSorterByDefault() bool {
+	if v.isUnknown {
+		// we assume the unknown version to be the latest version
+		return true
+	}
+	// x >= 4.0.13 AND x != 5.0.0-rc
+	if v.String() == "5.0.0-rc" {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (liuzix) [reply pkg/version/check.go:175]: There has been examples of users trying to use 5.0.0-rc in production, so I suggest that we be careful here.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**